lib/sycamore/sycamore/query/execution/operations.py

```diff
@@ -358,40 +360,45 @@ def math_operation(val1: int, val2: int, operator: str) -> Union[int, float]:
 # USAGE EXAMPLE
 # operations.llm_generate_operation("database of airplane incidents (NTSB)",
 # "How many incidents occurred between 1/1/2022 and 1/20/2022?", 13)
-def llm_generate_operation(client: OpenAI, question: str, result_description: str, result_data: Any, **kwargs) -> str:
-    if isinstance(result_data, DocSet):
-        text = f"Description: {result_description}\n"
-        for doc in result_data.take(**kwargs):
-            text += "\n"
-            if "entity" in doc.properties:
-                text += "Details: " + json.dumps(doc.properties["entity"], indent=2) + "\n"
-                props_dict = doc.properties["entity"]
-            else:
-                props_dict = {}
-
-            props_dict.update({p: doc.properties[p] for p in set(doc.properties) - set(BASE_PROPS)})
+def llm_generate_operation(
```

Comment:
While we're here, I think it would be good to start adding docstrings to methods. It's not so obvious from the name what this is supposed to do.